src/pybind/mgr/cephadm/service_discovery.py

```diff
@@ -0,0 +1,189 @@
+import cherrypy
+from cherrypy._cpserver import Server
+import json
+import logging
+import tempfile
+from typing import Dict, List, TYPE_CHECKING, cast, Collection
+
+from orchestrator import OrchestratorError
+from mgr_module import ServiceInfoT
+from mgr_util import verify_tls_files, build_url
+from cephadm.services.ingress import IngressSpec
+from cephadm.ssl_cert_utils import SSLCerts
+
+if TYPE_CHECKING:
+    from cephadm.module import CephadmOrchestrator
+
+
+def cherrypy_filter(record: logging.LogRecord) -> int:
+    blocked = [
+        'TLSV1_ALERT_DECRYPT_ERROR'
+    ]
+    msg = record.getMessage()
+    return not any([m for m in blocked if m in msg])
+
+
+logging.getLogger('cherrypy.error').addFilter(cherrypy_filter)
+cherrypy.log.access_log.propagate = False
+
+
+class ServiceDiscovery():
```

Comment:
Shouldn't this PR include a docs update to doc/cephadm/services/monitoring.rst?

Reply:
yes, sure. I'll try to add the needed docs.

Reply:
Yep the docs will need to evolve with the feature - however, things like how to get the PEM file is relevant with this PR and should be documented here.